lib/miq_automation_engine/engine/miq_ae_engine/miq_ae_builtin_method.rb

```diff
@@ -182,11 +185,15 @@ def self.detect_category(obj_name, prov_obj)
       when "orchestration_stack", "orchestration_stack_retire_task"
         CLOUD
       when "miq_request", "miq_provision", "vm_migrate_task", "vm_retire_task"
+        return PHYSICAL_INFRA if (prov_obj&.options || {})[:request_type] == 'provision_physical_server'
```

Comment:
Instead of checking the options hash, could we verify they sub-class of object?  Same for the `miq_provision_task` case below.  Event though these are different objects, one request and one task, maybe we could pull this logic into a `physical_server?` method here to DRY up the logic.

Reply:
Good point - turns out it's now even more simple because we've separated task name. For MiqRequest I've done it with case statement, please let me know if you find it readable enough. If not, I'll use auxilary function as you suggest.